product-scenarios/1-manage-users-roles/1.1-user-registration-to-an-application/1.1.1-user-registration-with-web-application-itself/1.1.1.1-provision-user-using-SCIM1.1/src/test/java/org/wso2/identity/scenarios/test/scim/UserProvisionSCIMTestCase.java

```diff
@@ -66,56 +52,53 @@ public void testInit() throws Exception {
+        rootObject.put(SCIMConstants.USER_NAME_ATTRIBUTE, SCIMConstants.USERNAME);
+        rootObject.put(SCIMConstants.PASSWORD_ATTRIBUTE, SCIMConstants.PASSWORD);
 
-        StringEntity entity = new StringEntity(rootObject.toString());
-        request.setEntity(entity);
+        HttpResponse response = IdentityScenarioUtil.sendPostRequestWithJSON(client, scimEndpoint, rootObject,
+                new Header[]{getBasicAuthzHeader(), getContentTypeApplicationJSONHeader()});
 
-        HttpResponse response = client.execute(request);
-        assertEquals(response.getStatusLine().getStatusCode(), 201, "User has not been created successfully");
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_CREATED, "User has not been created successfully");
 
-        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
-        EntityUtils.consume(response.getEntity());
+        JSONObject responseObj = IdentityScenarioUtil.getJSONFromResponse(response);
+        String usernameFromResponse = responseObj.get(SCIMConstants.USER_NAME_ATTRIBUTE).toString();
+        assertEquals(usernameFromResponse, SCIMConstants.USERNAME);
 
-        String usernameFromResponse = ((JSONObject) responseObj).get(USER_NAME_ATTRIBUTE).toString();
-        assertEquals(usernameFromResponse, USERNAME);
-
-        userId = ((JSONObject) responseObj).get(ID_ATTRIBUTE).toString();
+        userId = responseObj.get(SCIMConstants.ID_ATTRIBUTE).toString();
         assertNotNull(userId);
-
-        testDeleteUser();
     }
 
-    private void testDeleteUser() throws Exception {
+    @AfterClass(alwaysRun = true)
+    public void testDeleteUser() throws Exception {
 
         String userResourcePath =
-                getDeploymentProperties().getProperty(IS_HTTPS_URL) + SCIM_USERS_ENDPOINT + "/" + userId;
-        HttpDelete request = new HttpDelete(userResourcePath);
-        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
-        request.addHeader(HttpHeaders.CONTENT_TYPE, CONTENT_TYPE_APPLICATION_JSON);
-
-        HttpResponse response = client.execute(request);
-        assertEquals(response.getStatusLine().getStatusCode(), 200, "User has not been retrieved successfully");
-        EntityUtils.consume(response.getEntity());
-        userResourcePath = getDeploymentProperties().getProperty(IS_HTTPS_URL) + SCIM_USERS_ENDPOINT + "/" + userId;
-        HttpGet getRequest = new HttpGet(userResourcePath);
-        getRequest.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
-        getRequest.addHeader(HttpHeaders.CONTENT_TYPE, CONTENT_TYPE_APPLICATION_JSON);
-
-        response = client.execute(request);
-        assertEquals(response.getStatusLine().getStatusCode(), 404, "User has not been deleted successfully");
-        EntityUtils.consume(response.getEntity());
+                getDeploymentProperties().getProperty(IS_HTTPS_URL) + SCIMConstants.SCIM_USERS_ENDPOINT + "/" + userId;
```

Comment:
Construct the the SCIM basic URL during `BeforeClass`and hold it in a global variable